src/devices/wifi/nm-device-iwd.c

```diff
@@ -481,15 +504,24 @@ deactivate_async (NMDevice *device,
 {
 	NMDeviceIwd *self = NM_DEVICE_IWD (device);
 	NMDeviceIwdPrivate *priv = NM_DEVICE_IWD_GET_PRIVATE (self);
-	DeactivateContext *ctx;
+	GTask *task;
```

Comment:
until now, we didn't use GTask (at all). Obviously, iwd doesn't use glib at all, but compared to Connman, which also doesn't use GTask either (and prides itself of not using gobject). Is using this API a good idea? How would it look like without this API?

To me, GTask seems to provide features we don't need. @bengal, WDYT?

Reply:
I can easily change this to GSimpleAsyncResult. I'd never used either before but chose GTask because GSimpleAsyncResult is documented as deprecated in favour of GTask.

Reply:
OK. I think we should get rid of all GAsyncResult/GTask API as much as we can. Since NMDeviceClass.deactivate_async() follows this pattern, it bubbles all the way down. This should be refactored (will do separately). The patch is good for the moment.

Reply:
Ok! The only instances are in device/wwan and device/bluetooth so I assume this won't conflict with this patch and will be done afterwards.